modules/govuk/manifests/apps/draft_publicapi.pp

```diff
@@ -0,0 +1,50 @@
+# FIXME: This class needs better documentation as per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc
+class govuk::apps::draft_publicapi (
```

Comment:
I wonder if there is a way we can avoid the duplication between this and `publicapi`.

Could we maybe have a common `govuk::apps::publicapi` class and do something like:

```
govuk::apps::publicapi { 'draft-publicapi':
  backdrop_protocol => $backdrop_protocol,
  backdrop_host => $backdrop_host,
  ...
}
```

Similar to how we've defined the parts of `govuk::app`, for example [`govuk::app::package`](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/app/package.pp).

Reply:
How would that work with including the apps in the YAML files?

Reply:
I think we could have two classes `draft_publicapi` and `live_publicapi` which then just use the `publicapi` definition.

```puppet
class govuk::apps::draft_publicapi (...) {
  govuk::app::publicapi { 'draft-publicapi':
    backdrop_protocol => $backdrop_protocol,
    backdrop_host => $backdrop_host,
    ...
  }
}

class govuk::apps::live_publicapi (...) {
  govuk::app::publicapi { 'live-publicapi':
    backdrop_protocol => $backdrop_protocol,
    backdrop_host => $backdrop_host,
    ...
  }
}
```

Maybe it's too much refactoring for this PR, I'm just thinking that if we need to change something in the live one it might get missed in the draft one.